javascript/src/test/components/QuarterFilterForm.test.js

```diff
@@ -28,7 +32,71 @@ describe("QuarterFilterForm tests", () => {
     userEvent.click(submitButton);
 
     expect(upsertFilterMock).toHaveBeenCalledTimes(1);
-    expect(upsertFilterMock).toHaveBeenCalledWith(sampleFilterVal);
   });
+  test("set quarter filter format works with right format", async () => {
```

Comment:
For these 4 tests, verify that there is actual corresponding correct behavior, not just that the filter is called. For example, for the 3 tests with invalid values, verify that the error message is actually displayed on the page after submission. For the correct test, verify that the error message is not present after submission